skrifa/src/outline/glyf/hint/graphics_state/mod.rs

```diff
@@ -104,6 +104,10 @@ pub struct GraphicsState<'a> {
     ///
     /// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.h#L195>
     pub is_pedantic: bool,
+    /// Set to true when IUP has been executed in the horizontal direction.
+    pub did_iup_x: bool,
+    /// Set to true when IUP has been executed in the vertical direction.
+    pub did_iup_y: bool,
```

Comment:
Do we typically iup in both directions at the same time or is there a period when instructions could run when only one of these are set? - we have if statements for `back compat && did_iup_x && did_iup_y` and it's not clear to me if only one of them being set is a realistic state.

Basically why can't we have just `did_iup: bool,`

Reply:
Yes, IUP in x and y are different instructions that can be executed separately and modifying the behavior based on this is basically the whole idea behind “modern vertical only hinting” and the backward compatibility flag controls how we do this for fonts that contain bytecode that does full grid fitting.